src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java

```diff
@@ -123,7 +118,32 @@ static String getExternFileNameOrNull(String testFileName) {
     filesList.addAll(Arrays.asList(testMultifilePartailFiles));
     filesList.addAll(Arrays.asList(testPartialCrossModuleTypeImportsFiles));
     filesList.addAll(Arrays.asList(testOutputBaseFiles));
-    return filesList;
+
+    Stream<File> jsFiles = filesList.stream().filter(f -> !f.getName().endsWith(".zip"));
+    Stream<File> zipFiles =
+        filesList
+            .stream()
+            .filter(f -> f.getName().endsWith(".zip"))
+            .map(File::getPath)
+            .flatMap(
+                f -> {
+                  try (ZipFile zipFile = new ZipFile(f)) {
+                    return zipFile
+                        .stream()
+                        .filter(e -> !e.isDirectory())
+                        .filter(e -> e.getName().endsWith(".js"))
+                        .map(e -> f + "!/" + e.getName())
```

Comment:
some duplicated logic with DeclarationGenerator. Consider extracting to a helper.

Reply:
Done